atoma-proxy/src/sui/mod.rs

```diff
@@ -0,0 +1,232 @@
+    atoma_db_id: ObjectID,
+    toma_package_id: ObjectID,
+}
+
+impl Sui {
+    /// Constructor
+    pub async fn new(sui_config: &AtomaSuiConfig) -> Result<Self> {
+        // let keystore = FileBasedKeystore::new(&config.sui.sui_keystore_path().into())
+        //     .context("Failed to initialize keystore")
+        //     .expect("Failed to initialize keystore");
+        let sui_config_path = sui_config.sui_config_path();
+        let sui_config_path = Path::new(&sui_config_path);
+        let wallet_ctx = WalletContext::new(
+            sui_config_path,
+            sui_config.request_timeout(),
+            sui_config.max_concurrent_requests(),
+        )?;
+
+        Ok(Self {
+            wallet_ctx,
+            toma_wallet_id: None,
+            atoma_package_id: sui_config.atoma_package_id(),
+            atoma_db_id: sui_config.atoma_db(),
+            toma_package_id: sui_config.toma_package_id(),
+        })
+    }
+
+    #[instrument(level = "info", skip_all, fields(
+      endpoint = "acquire_new_stack_entry",
+      address = %self.wallet_ctx.active_address().unwrap()
+    ))]
+    pub async fn acquire_new_stack_entry(
+        &mut self,
+        task_small_id: u64,
+        num_compute_units: u64,
+        price: u64,
+    ) -> Result<i64> {
+        let client = self.wallet_ctx.get_client().await?;
+        let address = self.wallet_ctx.active_address()?;
+        let toma_wallet_id = self.get_or_load_toma_wallet_object_id().await?;
+
+        let tx = client
+            .transaction_builder()
+            .move_call(
+                address,
+                self.atoma_package_id,
+                "db",
+                "acquire_new_stack_entry",
+                vec![],
+                vec![
+                    SuiJsonValue::from_object_id(self.atoma_db_id),
+                    SuiJsonValue::from_object_id(toma_wallet_id),
+                    SuiJsonValue::new(task_small_id.to_string().into())?,
+                    SuiJsonValue::new(num_compute_units.to_string().into())?,
+                    SuiJsonValue::new(price.to_string().into())?,
+                    SuiJsonValue::from_object_id(SUI_RANDOMNESS_STATE_OBJECT_ID),
+                ],
+                None,
+                GAS_BUDGET,
+                None,
+            )
+            .await?;
+
+        info!("Submitting acquire new stack entry transaction...");
+        let tx = self.wallet_ctx.sign_transaction(&tx);
+        let response = self.wallet_ctx.execute_transaction_must_succeed(tx).await;
+
+        info!(
+            "Acquire new stack entry transaction submitted successfully. Transaction digest: {:?}",
+            response.digest
+        );
+        response
+            .events
+            .and_then(|event| {
+                event.data.get(0).and_then(|event| {
+                    event
+                        .parsed_json
+                        .get("selected_node_id")
+                        .and_then(|node_id| node_id.as_i64())
```

Comment:
I think this might not work, because the `u64` types in Sui Move, are parsed into `String` when you extract a `SuiJsonValue` object. For this reason, I believe you need to first extract a str and only then parse it into `u64` (I do recommend doing it for `u64` and not `i64`, because of future potential parsing issues).

Reply:
yeah, you are right